float32 angleX
float32 angleY
float32 rotation
